File: src/types/maintenance.ts
import { Room } from './room';
import { Employee } from './employee';

export type MaintenanceStatus = 'Pending' | 'InProgress' | 'Completed' | 'Cancelled';
export type MaintenancePriority = 'Low' | 'Medium' | 'High' | 'Emergency';
export type MaintenanceCategory = 'Plumbing' | 'Electrical' | 'HVAC' | 'General' | 'Structural' | 'Furniture' | 'Appliance' | 'Other';

export interface Maintenance {
    id: number;
    roomId: number;
    employeeId: number;
    title: string;
    description: string;
    status: MaintenanceStatus;
    priority: MaintenancePriority;
    category: MaintenanceCategory;
    scheduledDate?: string;
    completedDate?: string;
    resolution?: string;
    cost?: number;
    isUrgent: boolean;
    createdAt: string;
    lastUpdated?: string;
    room?: Room;
    employee?: Employee;
}

export interface MaintenanceCreate {
    roomId: number;
    employeeId: number;
    title: string;
    description: string;
    priority: MaintenancePriority;
    category: MaintenanceCategory;
    scheduledDate?: string;
    isUrgent: boolean;
}

export type MaintenanceUpdate = Partial<MaintenanceCreate> & {
    status?: MaintenanceStatus;
    completedDate?: string;
    resolution?: string;
    cost?: number;
}; 

File: Front/SmartStayHaven.API/Models/Maintenance.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartStayHaven.API.Models
{
    public class Maintenance
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int RoomId { get; set; }

        [Required]
        public int EmployeeId { get; set; }

        [Required]
        public string Title { get; set; } = string.Empty;

        [Required]
        public string Description { get; set; } = string.Empty;

        [Required]
        public string Status { get; set; } = "Pending"; // Pending, InProgress, Completed, Cancelled

        [Required]
        public string Priority { get; set; } = "Medium"; // Low, Medium, High, Emergency

        [Required]
        public string Category { get; set; } = string.Empty; // Plumbing, Electrical, HVAC, General, etc.

        public DateTime? ScheduledDate { get; set; }

        public DateTime? CompletedDate { get; set; }

        public string? Resolution { get; set; }

        public decimal? Cost { get; set; }

        public bool IsUrgent { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? LastUpdated { get; set; }

        // Navigation properties
        public Room? Room { get; set; }
        public Employee? Employee { get; set; }
    }
} 